eng/versioning/verify_release_set.ps1

```diff
@@ -0,0 +1,97 @@
+param(
+    [Parameter(Mandatory=$true)]
+    [System.String] $ServiceDirectory,
+    # ArtifactsList will be using ('${{ convertToJson(parameters.Artifacts) }}' | ConvertFrom-Json | Select-Object name, groupId)
+    [Parameter(Mandatory=$true)]
+    [array] $ArtifactsList
+)
+
+$resultsTime = [diagnostics.stopwatch]::StartNew()
+
+# Given the service directory and a list of libraries to be released verify that the list
+# of libraries has full transitive closure. This means that we have to look at each library's
+# pom file's current dependencies and ensure that each dependency is contained within the
+# list to be released.
+$script:FoundError = $false
+$libraryReleaseList = @()
+$missingLibraries = @()
+function Write-Error-With-Color([string]$msg)
+{
+    Write-Host "$($msg)" -ForegroundColor Red
+}
+
+Write-Host "ServiceDirectory=$($ServiceDirectory)"
+Write-Host "ArtifactsList:"
+$ArtifactsList | Format-Table -Property Name, GroupId | Out-String | Write-Host
+foreach($artifact in $ArtifactsList) {
+    $libraryReleaseList += ,($artifact.groupId + ":" + $artifact.name)
+}
+
+foreach($artifact in $ArtifactsList) {
+    $script:FoundPomFile = $false
+    $inputGroupId = $artifact.groupId
+    $inputArtifactId = $artifact.name
+
+    # It's unfortunate that we have to find the POM file this way for a given group/artifact.
+    # The reason is because an sdk/<area> should have subdirectories named for each artifact but
+    # that's not always the case so we need to discover.
+    Get-ChildItem -Path $ServiceDirectory -Filter pom*.xml -Recurse -File | ForEach-Object {
```

Comment:
Generally better to prefer a foreach(...) loop over the ForEach-Object pipeline cmdlet.

Reply:
Already committed.